src/main/java/co/inventorsoft/academy/springBootTask/service/impl/TournamentBracketServiceImpl.java

```diff
@@ -0,0 +1,93 @@
+package co.inventorsoft.academy.springBootTask.service.impl;
+
+import co.inventorsoft.academy.springBootTask.domain.dto.TeamDto;
+import co.inventorsoft.academy.springBootTask.domain.dto.TournamentBracketDto;
+import co.inventorsoft.academy.springBootTask.domain.entity.TournamentBracket;
+import co.inventorsoft.academy.springBootTask.domain.mapper.TournamentBracketMapper;
+import co.inventorsoft.academy.springBootTask.repository.TournamentBracketRepository;
+import co.inventorsoft.academy.springBootTask.service.TeamService;
+import co.inventorsoft.academy.springBootTask.service.TournamentBracketService;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Service
+@Slf4j
+@RequiredArgsConstructor
+public class TournamentBracketServiceImpl implements TournamentBracketService {
+
+    private final TournamentBracketRepository tournamentBracketRepository;
+
+    TournamentBracketDto tbDto;
+    private final TeamService teamService;
```

Comment:
1) Group dependencies that will be injected together.
2) You chose bad approach to inject DB service into another DB service. You will get very easy to circular dependency issues.